system/Database/BaseConnection.php

```diff
@@ -194,7 +195,14 @@ abstract class BaseConnection implements ConnectionInterface
      * @var array
      */
     protected $failover = [];
-
+    
+    /**
+     * Application name for Postgres DSN.
+     *
+     * @var array
+     */
+    protected $application_name = [];
```

Comment:
Why is this an array? It seems to be a string.
In any case, add the type for new properties.